### ISSUE TITLE
Investigate shadowJar + gradle UP-TO-DATE check

### ISSUE THREAD

**Issue by trask** [open-telemetry/opentelemetry-java-instrumentation#3500]
Spurred by https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/3474#discussion_r663682249, I just re-tried the UP-TO-DATE failure from #3269:
1. remove `inputs.file(shadowJar.get().archiveFile)` from `javaagent/build.gradle.kts`
2. run `./gradlew :javaagent:test`, passes
3. introduce bug in `ExtensionClassLoader.java` by commenting out:
```
    if (extensions.isEmpty()) {
      return parent;
    }
```
4. run `./gradlew :javaagent:test`, still passes (because it thinks it's up-to-date)

Maybe shadowJar is not telling gradle about its output? Would be good to know why and add comment(s) to the gradle configuration(s) to explain why the extra inputs are needed for future readers.

**Comment by ryandens** [comment]
👋 I was taking a look at this issue and it seems like this was resolved by #4794 because the test task now uses a `CommandLineArgumentProvider` connected to the javaagent shadowjar input file rather than consuming the file name as a String.

I tried the reproduction steps described in this issue, and while I did confirm that `javaagent:test` forces the rebuild of `javaagent-tooling` when the changes are made and re-runs the test task (because the tasks are not up to date) the tests did pass. As a different test, I changed [this line] to default to `false` rather than `true` and demonstrated that `ClassLoadingTest` now fails. 

I think this should be closed.
